### PR TITLE
Various Process enhancements

### PR DESCRIPTION
**Pull request by jschueller**
- Add GaussianProcess::getMarginal 
- Add ProcessSample::drawMarginal interpolate option 
- Add Field._repr_html_ 
-  Fix http://trac.openturns.org/ticket/954

### CODE REVIEW

**Comment by regislebrun** [comment]
Not 100% sure that CovarianceModel::getMarginal() is correct (what about the management of active parameters of a  TensorizedCovarianceModel?), otherwise +1 (and thanks for the last commit)

**Comment by jschueller** [comment]
Yes, it should be reimplemented for TensorizedCovarianceModel, I can mark it as notyetimplemented.

Also I noticed TensorizedCovarianceModel(CovarianceModelCollection) initializes with a scale of 1.0, why not take the scale of each cov model in the list like it's done for the amplitude ?

**Comment by regislebrun** [comment]
It is because TensorizedCovarianceModel is here to stack p covariance 1D models sharing the same input dimension to form a p dimensional covariance model. So it is not possible to pick one scale here and another here. Instead of taking the first model scale as the tensorized scale, a default value is used. This behavior can be change if you want.

**Comment by jschueller** [comment]
no, it's ok

File: lib/src/Uncertainty/Process/CompositeProcess.cxx [hunk 1]
```diff
@@ -129,7 +129,16 @@ TimeSeries CompositeProcess::getFuture(const UnsignedInteger stepNumber) const
   /* TimeGrid associated with the possible future */
   const Scalar timeStep = timeGrid.getStep();
   const RegularGrid futurTimeGrid(timeGrid.getEnd(), timeStep, stepNumber);
-  return TimeSeries(futurTimeGrid, function_(antecedent_.getFuture(stepNumber).getValues()));
+  FieldFunction function(function_);
+  try
+  {
+    function.setInputMesh(futurTimeGrid);
```
Comment (dbarbier) [inline lib/src/Uncertainty/Process/CompositeProcess.cxx]: Even if this is useless here, I suggest to also call setOutputMesh.

Comment (jschueller) [reply lib/src/Uncertainty/Process/CompositeProcess.cxx]: Did you remove this comment ?

Comment (dbarbier) [reply lib/src/Uncertainty/Process/CompositeProcess.cxx]: No

Comment (jschueller) [reply lib/src/Uncertainty/Process/CompositeProcess.cxx]: ok, it's just that it appeared grayed out

Comment (jschueller) [reply lib/src/Uncertainty/Process/CompositeProcess.cxx]: ok, done

**Comment by dbarbier** [comment]
Thanks, but I am still uncomfortable with setInputMesh, sorry about that.

In fact, ValueFunction and VertexValueFunction classes are designed to have the same mesh as input and output meshes; here we add FieldFunctionImplementation::set{In,Out}putMesh methods, and they are defined only within these two classes.  Maybe we should add a single method setInputOutputMesh instead?

**Comment by jschueller** [comment]
I've no strong opinion about one or the other. I'd say it depends on whether you prefer to reflect the actual storage of the meshes or the fact that they share input/output meshes.

File: lib/src/Base/Stat/CovarianceModelImplementation.cxx [hunk 2]
```diff
@@ -791,9 +791,63 @@ Bool CovarianceModelImplementation::isDiagonal() const
...
+  if (outputDimension_ > 1)
+  {
+    const UnsignedInteger oldOutputDimension = outputDimension_;
+    result.getImplementation()->outputDimension_ = indices.getSize();
+    Point newAmplitude;
+    for (UnsignedInteger i = 0; i < indices.getSize(); ++ i)
+    {
+      newAmplitude.add(amplitude_[indices[i]]);
+    }
+    result.getImplementation()->amplitude_ = newAmplitude;
```
Comment (sofianehaddad) [inline lib/src/Base/Stat/CovarianceModelImplementation.cxx]: I think we should use setter. Some models have to calibrate their internal parameters

File: lib/src/Base/Stat/CovarianceModelImplementation.cxx [hunk 3]
```diff
@@ -791,9 +791,63 @@ Bool CovarianceModelImplementation::isDiagonal() const
...
+      {
+        // amplitude (assumes ascending order)
+        newActiveParameters.add(inputDimension_ + counter);
+        ++ counter;
+      }
+      else
+      {
+        // others
+        newActiveParameters.add(activeParameters[i] - oldOutputDimension - indices.getSize());
+      }
```
Comment (sofianehaddad) [inline lib/src/Base/Stat/CovarianceModelImplementation.cxx]: No. others might contain output correlation coefficients. In that case, the code here is wrong.

Comment (jschueller) [reply lib/src/Base/Stat/CovarianceModelImplementation.cxx]: Yes, I dont think we can resolve this correctly here, I'll rather just resize the activeParameters.

File: lib/src/Base/Stat/CovarianceModelImplementation.cxx (hunk 2 above)
Comment (jschueller) [reply lib/src/Base/Stat/CovarianceModelImplementation.cxx]: ok

File: lib/src/Base/Stat/CovarianceModelImplementation.cxx (hunk 3 above)
Comment (jschueller) [reply lib/src/Base/Stat/CovarianceModelImplementation.cxx]: ok

**Comment by regislebrun** [comment]
IMO we should add a property to FieldFunction to check if it acts pointwise (isPointWise() true for ValueFunction and VertexValueFunction, false otherwise), restrict CompositeProcess::getFuture() to the case where isPointWise() is true and call only setInputMesh().

**Comment by jschueller** [comment]
@regislebrun , @dbarbier I added the FieldFunction::isPointWise property

File: lib/src/Base/Func/FieldFunctionImplementation.cxx [hunk 4]
```diff
@@ -138,14 +141,21 @@ Mesh FieldFunctionImplementation::getInputMesh() const
...
+  outputMesh_ = outputMesh;
 }
 
 Mesh FieldFunctionImplementation::getOutputMesh() const
 {
   return outputMesh_;
 }
 
+Bool FieldFunctionImplementation::isPointWise() const
+{
```
Comment (sofianehaddad) [inline lib/src/Base/Func/FieldFunctionImplementation.cxx]: why "w" is in capital letters ? isPointwise or isActingPointwise?

Comment (jschueller) [reply lib/src/Base/Func/FieldFunctionImplementation.cxx]: I agree, isActingPointwise is better

**Pull request merged**